plugins/by-name/openscad/default.nix

```diff
@@ -1,81 +1,170 @@
+    load_snippets = true;
+    fuzzy_finder = "fzf";
+    cheatsheet_window_blend = 15;
+    auto_open = true;
   };
+  settingsDescription = settingsOptionDescription { inherit name globalPrefix; };
+  callSetup = false;
 
-  config =
-    let
-      cfg = config.plugins.openscad;
-      fuzzyFinder = if (cfg.fuzzyFinder == null) then defaultFuzzyFinder else cfg.fuzzyFinder;
-    in
-    mkIf cfg.enable {
-      extraPlugins =
-        with pkgs.vimPlugins;
-        [ cfg.package ]
-        ++ (optional (fuzzyFinder == "skim") skim-vim)
-        ++ (optional (fuzzyFinder == "fzf") fzf-vim);
-
-      extraConfigLua = ''
-        require('openscad')
-      '';
-
-      globals = mkMerge [
-        {
-          openscad_fuzzy_finder = cfg.fuzzyFinder;
-          openscad_cheatsheet_window_blend = cfg.cheatsheetWindowBlend;
-          openscad_load_snippets = cfg.loadSnippets;
-        }
-        (mkIf cfg.keymaps.enable {
-          openscad_default_mappings = true;
-          openscad_cheatsheet_toggle_key = cfg.keymaps.cheatsheetToggle;
-          openscad_help_trig_key = cfg.keymaps.helpTrigger;
-          openscad_help_manual_trig_key = cfg.keymaps.helpManualTrigger;
-          openscad_exec_openscad_trig_key = cfg.keymaps.execOpenSCADTrigger;
-          openscad_top_toggle = cfg.keymaps.topToggle;
-        })
-      ];
-    };
+  extraOptions = {
+    fuzzyFinderPlugin =
+      let
+        optionName = "`plugins.${name}.settings.fuzzy_finder`";
+
+        fuzzyFinder = config.plugins.openscad.settings.fuzzy_finder;
+
+        defaultPluginName =
+          {
+            skim = "skim-vim";
+            fzf = "fzf-vim";
+          }
+          .${fuzzyFinder} or null;
+
+        default =
+          # If the user has not set `settings.fuzzy_finder`, do not pre-install a fuzzy-finder by default.
+          if fuzzyFinder == null then
+            null
+          # Else, the value of `settings.fuzzy_finder` should be one of the supported options
+          # (`skim` or `fzf`), else he has to provide a value (`null` or a package) to `fuzzyFinderPlugin`.
+          else if defaultPluginName == null then
+            throw ''
+              We cannot automatically select a fuzzy finder plugin from the value given to `${optionName}`: "${fuzzyFinder}".
+              Please, explicitly provide a value to the `plugins.openscad.fuzzyFinderPlugin`:
+                - Either the package for the fuzzy finder plugin to be installed
+                - or `null` if you do not want a plugin to be installed.
+            ''
+          # Case where we automatically select the default plugin to install.
+          else
+            [
+              "vimPlugins"
+              defaultPluginName
+            ];
+      in
+      lib.mkPackageOption pkgs "fuzzy finder" {
+        nullable = true;
+        inherit default;
+        extraDescription = ''
+          If ${optionName} is set explicitly, this option will default to
+        '';
+      };
```

Comment:
I'd much rather have a `literalMD` type `defaultText` than explain the default in the `description`

Maybe `mkPackageOption` supports `defaultText`? Looks like it does not: https://noogle.dev/f/lib/mkPackageOption

Alternatively, we can define the option manually, or just do `// { defaultText = ... }`

```nix
lib.mkPackageOption pkgs "fuzzy finder" {
  nullable = true;
  inherit default;
}
// {
  defaultText = literalMD ''
    If `${optionName}` is set explicitly, then either
    ${concatStringsSep ", " (attrValues defaultPluginNames)}
    as appropriate. Otherwise, `null`.
  '';
}
```

Reply:
I think you reviewed just before I pushed something in this style.

Reply:
We had the same idea at the same time haha. Any feedback on the wording ?

Reply:
![image](https://github.com/user-attachments/assets/bbe9e82b-7da7-41fd-8c85-7f71c2722f47)

Reply:
Since `settings.fuzzy_finder` defaults to null, it is actually better not mentioning "explicit" definitions. So this is perfect.

I wasn't sure how a multi-line value would render, but it looks great in your screenshot.